### PR TITLE
State Tree Linking

### LINKED ISSUE #484: Follow ObjectIDs in state view
I assume the ability to follow object IDs was lost during proto change?

### PR DESCRIPTION
**Pull request by pmuetschard**
Closes #484

### CODE REVIEW

File: gapis/resolve/state_tree.go [hunk 1]
```diff
@@ -242,7 +242,7 @@ func (n *stn) buildChildren(ctx context.Context, tree *stateTree) {
 				children = append(children, &stn{
 					name:           fmt.Sprintf("[%d - %d]", n.subgroupOffset+s, n.subgroupOffset+e-1),
 					value:          reflect.ValueOf(slice.ISlice(s, e, tree.state.MemoryLayout)),
-					path:           n.path,
+					path:           path.NewArrayIndex(n.subgroupOffset+s, n.path),
```
Comment (ben-clayton) [inline gapis/resolve/state_tree.go]: Isn't this assigning the path to the first element in the slice? Shouldn't it be a subslice of the slice?

Also, don't the tests need updating?

File: gapis/resolve/state_tree_test.go [hunk 2]
```diff
@@ -491,7 +491,7 @@ func TestStateTreeNode(t *testing.T) {
 			&service.StateTreeNode{
 				NumChildren:    10,
 				Name:           "[10 - 19]",
-				ValuePath:      rootPath.Field("ReferenceB").Field("Array").Path(),
+				ValuePath:      rootPath.Field("ReferenceB").Field("Array").Slice(10, 19).Path(),
```
Comment (ben-clayton) [inline gapis/resolve/state_tree_test.go]: Now that these paths are unique, you can probably kill the `flags    int` field from the test structure. The only reason this existed was that there was a many-to-1 mapping of value path -> STN. Now I believe they're all 1:1.

File: gapis/resolve/state_tree.go (hunk 1 above)
Comment (pmuetschard) [inline gapis/resolve/state_tree.go]: Done and done. <sub>I'm so <sub>ashamed...</sub></sub>

File: gapis/resolve/state_tree_test.go (hunk 2 above)
Comment (pmuetschard) [inline gapis/resolve/state_tree_test.go]: Done.

**Pull request merged**